failed to setup livekit-plugin-browser locally

## Bug Report

### Description
livekit-plugin-browser not correctly setting up in local dev

### To Reproduce
Try to install the plugins with the following command inside `livekit-plugins`
```bash
./install_plugins_editable.sh
```

### Environment
- OS: Macos 15
- Python Version: Python 3.12.0

### Error Logs
```bash
 Cloning into 'imgui-src'...
      error: RPC failed; curl 18 transfer closed with outstanding read data remaining
      error: 3234 bytes of body are still expected
      fetch-pack: unexpected disconnect while reading sideband packet
      fatal: early EOF
      fatal: fetch-pack: invalid index-pack output
      Cloning into 'imgui-src'...
      error: RPC failed; curl 18 transfer closed with outstanding read data remaining
      error: 5203 bytes of body are still expected
      fetch-pack: unexpected disconnect while reading sideband packet
      fatal: early EOF
      fatal: fetch-pack: invalid index-pack output
      Cloning into 'imgui-src'...
      error: RPC failed; curl 18 transfer closed with outstanding read data remaining
      error: 8277 bytes of body are still expected
      fetch-pack: unexpected disconnect while reading sideband packet
      fatal: early EOF
      fatal: fetch-pack: invalid index-pack output
      Had to git clone more than once: 3 times.
      CMake Error at imgui-subbuild/imgui-populate-prefix/tmp/imgui-populate-gitclone.cmake:50 (message):
        Failed to clone repository: 'https://github.com/ocornut/imgui'
      
      
      make[2]: *** [imgui-populate-prefix/src/imgui-populate-stamp/imgui-populate-download] Error 1
      make[1]: *** [CMakeFiles/imgui-populate.dir/all] Error 2
      make: *** [all] Error 2
      CMake Error at /opt/homebrew/Cellar/cmake/3.30.1/share/cmake/Modules/FetchContent.cmake:1918 (message):
        Build step for imgui failed: 2
      Call Stack (most recent call first):
        /opt/homebrew/Cellar/cmake/3.30.1/share/cmake/Modules/FetchContent.cmake:1609 (__FetchContent_populateSubbuild)
        /opt/homebrew/Cellar/cmake/3.30.1/share/cmake/Modules/FetchContent.cmake:2145:EVAL:2 (__FetchContent_doPopulation)
        /opt/homebrew/Cellar/cmake/3.30.1/share/cmake/Modules/FetchContent.cmake:2145 (cmake_language)
        /opt/homebrew/Cellar/cmake/3.30.1/share/cmake/Modules/FetchContent.cmake:2384 (__FetchContent_Populate)
        src/CMakeLists.txt:21 (FetchContent_MakeAvailable)
      
      
      -- Configuring incomplete, errors occurred!
      Traceback (most recent call last):
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/command/editable_wheel.py", line 139, in run
          self._create_wheel_file(bdist_wheel)
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/command/editable_wheel.py", line 340, in _create_wheel_file
          files, mapping = self._run_build_commands(dist_name, unpacked, lib, tmp)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/command/editable_wheel.py", line 263, in _run_build_commands
          self._run_build_subcommands()
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/command/editable_wheel.py", line 290, in _run_build_subcommands
          self.run_command(name)
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 995, in run_command
          super().run_command(command)
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
          cmd_obj.run()
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/command/build_ext.py", line 99, in run
          _build_ext.run(self)
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
          self.build_extensions()
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 476, in build_extensions
          self._build_extensions_serial()
        File "/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 502, in _build_extensions_serial
          self.build_extension(ext)
        File "<string>", line 65, in build_extension
        File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py", line 571, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['cmake', '/Users/mbuke_repo/Documents/projects/opensource/agents/livekit-plugins/livekit-plugins-browser', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/tmp21wscgdi.build-lib', '-DPYTHON_EXECUTABLE=/Users/mbuke_repo/Documents/projects/opensource/agents/.venv/bin/python3.12', '-DCMAKE_BUILD_TYPE=Release']' returned non-zero exit status 1.
      /private/var/folders/37/pzqgw0dj5bd4tl745_39c9hh0000gp/T/pip-build-env-s00pa62d/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py:973: _DebuggingTips: Problem in editable installation.
      !!
```

--- a/.changeset/fix-imgui-setup.md
+++ b/.changeset/fix-imgui-setup.md
@@ -1,0 +1,5 @@
+---
+"livekit-plugins-browser": patch
+---
+
+fix: fix `imgui` setup

--- a/livekit-plugins/livekit-plugins-browser/src/CMakeLists.txt
+++ b/livekit-plugins/livekit-plugins-browser/src/CMakeLists.txt
@@ -11,8 +11,21 @@ set(GLFW_INSTALL OFF CACHE BOOL "" FORCE)
 FetchContent_Declare(glfw GIT_REPOSITORY https://github.com/glfw/glfw.git GIT_TAG 3.4)
 FetchContent_MakeAvailable(glfw)
 
-FetchContent_Declare(imgui GIT_REPOSITORY https://github.com/ocornut/imgui GIT_TAG origin/docking)
+FetchContent_Declare(
+  imgui 
+  GIT_REPOSITORY https://github.com/ocornut/imgui 
+  GIT_TAG origin/docking
+  GIT_SHALLOW TRUE
+)
 FetchContent_GetProperties(imgui)
+if(NOT imgui_POPULATED)
+    FetchContent_Populate(imgui)
+    execute_process(
+        COMMAND git fetch --unshallow
+        WORKING_DIRECTORY ${imgui_SOURCE_DIR}
+    )
+endif()
+
 FetchContent_MakeAvailable(imgui)
 file(GLOB IMGUI_SOURCES ${imgui_SOURCE_DIR}/*.cpp)
 add_library(imgui STATIC ${IMGUI_SOURCES}

--- a/livekit-plugins/livekit-plugins-browser/src/CMakeLists.txt
+++ b/livekit-plugins/livekit-plugins-browser/src/CMakeLists.txt
@@ -19,10 +19,6 @@ FetchContent_Declare(
 )
 FetchContent_GetProperties(imgui)
 FetchContent_Populate(imgui)
-execute_process(
-    COMMAND git fetch --unshallow
-    WORKING_DIRECTORY ${imgui_SOURCE_DIR}
-)
 
 FetchContent_MakeAvailable(imgui)
 file(GLOB IMGUI_SOURCES ${imgui_SOURCE_DIR}/*.cpp)

--- a/livekit-plugins/livekit-plugins-browser/src/CMakeLists.txt
+++ b/livekit-plugins/livekit-plugins-browser/src/CMakeLists.txt
@@ -18,13 +18,11 @@ FetchContent_Declare(
   GIT_SHALLOW TRUE
 )
 FetchContent_GetProperties(imgui)
-if(NOT imgui_POPULATED)
-    FetchContent_Populate(imgui)
-    execute_process(
-        COMMAND git fetch --unshallow
-        WORKING_DIRECTORY ${imgui_SOURCE_DIR}
-    )
-endif()
+FetchContent_Populate(imgui)
+execute_process(
+    COMMAND git fetch --unshallow
+    WORKING_DIRECTORY ${imgui_SOURCE_DIR}
+)
 
 FetchContent_MakeAvailable(imgui)
 file(GLOB IMGUI_SOURCES ${imgui_SOURCE_DIR}/*.cpp)

